### PR TITLE
Add shortcut keys for layouts

### PR DESCRIPTION
**Pull request by cinqmilleans**
Add keyboard shortcuts for layouts: ctrl+shift+1/6.
Reorder layouts in menu, from the smallest to the biggest.
#4682

### CODE REVIEW

File: Files/Strings/en-US/Resources.resw [hunk 1]
```diff
@@ -1558,7 +1558,7 @@
     <value>Folders</value>
   </data>
   <data name="NavToolbarDetailsView.ToolTipService.ToolTip" xml:space="preserve">
-    <value>Details View</value>
+    <value>Details View (Ctrl+Shift+2)</value>
```
Comment (yaichenbaum) [inline Files/Strings/en-US/Resources.resw:1561]: Isn't the shortcut automatically displayed in the tooltip?

File: Files/Helpers/ContextFlyoutItemHelper.cs [hunk 2]
```diff
@@ -97,60 +98,66 @@ public static List<ContextMenuFlyoutItemViewModel> GetBaseLayoutMenuItems(Curren
...
-                            Text = "BaseLayoutContextFlyoutGridViewLarge/Text".GetLocalized(),
-                            Glyph = "\uE739",
+                            Text = "BaseLayoutContextFlyoutColumn/Text".GetLocalized(),
+                            Glyph = "\uF115",
+                            GlyphFontFamilyName = "CustomGlyph",
                             ShowInRecycleBin = true,
-                            Command =  currentInstanceViewModel.FolderSettings.ToggleLayoutModeGridViewLarge,
+                            Command = currentInstanceViewModel.FolderSettings.ToggleLayoutModeColumnView,
                             CommandParameter = true,
+                            KeyboardAccelerator = new KeyboardAccelerator{Key = VirtualKey.Number1, Modifiers = VirtualKeyModifiers.Control | VirtualKeyModifiers.Shift}
```
Comment (yaichenbaum) [inline Files/Helpers/ContextFlyoutItemHelper.cs:110]:
```suggestion
                            KeyboardAccelerator = new KeyboardAccelerator{Key = VirtualKey.Number1, Modifiers = VirtualKeyModifiers.Control | VirtualKeyModifiers.Shift, IsEnabled=False}
```

Can you make this change to the rest of them as well?

File: Files/UserControls/NavigationToolbar.xaml [hunk 3]
```diff
@@ -1061,6 +1061,25 @@
...
+                                        x:Uid="NavToolbarColumnView"
+                                        Width="36"
+                                        Height="32"
+                                        MinWidth="0"
+                                        Padding="0"
+                                        HorizontalContentAlignment="Center"
+                                        VerticalContentAlignment="Center"
+                                        Command="{x:Bind ToggleLayoutModeColumnView}"
+                                        CommandParameter="{xh:SystemTypeToXaml Bool=True}"
+                                        CornerRadius="4"
```
Comment (yaichenbaum) [inline Files/UserControls/NavigationToolbar.xaml:1074]:
```suggestion
```

**Review by yair100 — CHANGES_REQUESTED** [review]

File: Files/Strings/en-US/Resources.resw (hunk 1 above)
Comment (yaichenbaum) [reply Files/Strings/en-US/Resources.resw:1561]: @cinqmilleans

**Comment by cinqmilleans** [comment]
@yair100 This PR has become too complicated because it covers portions of code that have changed a lot. I propose to reapply the changes from the current state. I need to know what to do with the tooltips (see my post above).

**Comment by yaichenbaum** [comment]
@cinqmilleans My question above was if it's necessary to add the shortcuts to the tooltip since they should be added automatically regardless.

**Review by yair100 — DISMISSED** [review]

**Comment by yaichenbaum** [comment]
@cinqmilleans Can you resolve the merge conflicts?

**Review by winston-de — DISMISSED** [review]

File: Files/UserControls/NavigationToolbar.xaml [hunk 4]
```diff
@@ -1091,7 +1091,7 @@
                                         CornerRadius="{StaticResource ControlCornerRadius}"
                                         IsChecked="{x:Bind LayoutModeInformation, Converter={StaticResource LayoutModeConverter}, Mode=OneWay, ConverterParameter=DetailsInfo}"
                                         Style="{StaticResource ToggleButtonRevealStyle}"
-                                        ToolTipService.ToolTip="Details View">
+                                        ToolTipService.ToolTip="Details View (Ctrl+Shift+2)">
```
Comment (yaichenbaum) [inline Files/UserControls/NavigationToolbar.xaml:1094]:
```suggestion
                                        ToolTipService.ToolTip="Details View (Ctrl+Shift+1)">
```

File: Files/UserControls/NavigationToolbar.xaml [hunk 5]
```diff
@@ -1107,7 +1107,7 @@
                                         CornerRadius="{StaticResource ControlCornerRadius}"
                                         IsChecked="{x:Bind LayoutModeInformation, Converter={StaticResource LayoutModeConverter}, Mode=OneWay, ConverterParameter=TilesInfo}"
                                         Style="{StaticResource ToggleButtonRevealStyle}"
-                                        ToolTipService.ToolTip="Tiles View">
+                                        ToolTipService.ToolTip="Tiles View (Ctrl+Shift+3)">
```
Comment (yaichenbaum) [inline Files/UserControls/NavigationToolbar.xaml:1110]:
```suggestion
                                        ToolTipService.ToolTip="Tiles View (Ctrl+Shift+2)">
```

File: Files/UserControls/NavigationToolbar.xaml [hunk 6]
```diff
@@ -1125,7 +1125,7 @@
                                         CornerRadius="{StaticResource ControlCornerRadius}"
                                         IsChecked="{x:Bind LayoutModeInformation, Converter={StaticResource LayoutModeConverter}, Mode=OneWay, ConverterParameter=SmallGridInfo}"
                                         Style="{StaticResource ToggleButtonRevealStyle}"
-                                        ToolTipService.ToolTip="Grid View (Small)">
+                                        ToolTipService.ToolTip="Grid View (Small) (Ctrl+Shift+4)">
```
Comment (yaichenbaum) [inline Files/UserControls/NavigationToolbar.xaml:1128]:
```suggestion
                                        ToolTipService.ToolTip="Grid View (Small) (Ctrl+Shift+3)">
```

File: Files/UserControls/NavigationToolbar.xaml [hunk 7]
```diff
@@ -1141,7 +1141,7 @@
                                         CornerRadius="{StaticResource ControlCornerRadius}"
                                         IsChecked="{x:Bind LayoutModeInformation, Converter={StaticResource LayoutModeConverter}, Mode=OneWay, ConverterParameter=MediumGridInfo}"
                                         Style="{StaticResource ToggleButtonRevealStyle}"
-                                        ToolTipService.ToolTip="Grid View (Medium)">
+                                        ToolTipService.ToolTip="Grid View (Medium) (Ctrl+Shift+5)">
```
Comment (yaichenbaum) [inline Files/UserControls/NavigationToolbar.xaml:1144]:
```suggestion
                                        ToolTipService.ToolTip="Grid View (Medium) (Ctrl+Shift+4)">
```

File: Files/UserControls/NavigationToolbar.xaml [hunk 8]
```diff
@@ -1157,7 +1157,7 @@
                                         CornerRadius="{StaticResource ControlCornerRadius}"
                                         IsChecked="{x:Bind LayoutModeInformation, Converter={StaticResource LayoutModeConverter}, Mode=OneWay, ConverterParameter=LargeGridInfo}"
                                         Style="{StaticResource ToggleButtonRevealStyle}"
-                                        ToolTipService.ToolTip="Grid View (Large)">
+                                        ToolTipService.ToolTip="Grid View (Large) (Ctrl+Shift+6)">
```
Comment (yaichenbaum) [inline Files/UserControls/NavigationToolbar.xaml:1160]:
```suggestion
                                        ToolTipService.ToolTip="Grid View (Large) (Ctrl+Shift+5)">
```

File: Files/UserControls/NavigationToolbar.xaml [hunk 9]
```diff
@@ -1173,7 +1173,7 @@
                                         CornerRadius="{StaticResource ControlCornerRadius}"
                                         IsChecked="{x:Bind LayoutModeInformation, Converter={StaticResource LayoutModeConverter}, Mode=OneWay, ConverterParameter=ColumnInfo}"
                                         Style="{StaticResource ToggleButtonRevealStyle}"
-                                        ToolTipService.ToolTip="Column View">
+                                        ToolTipService.ToolTip="Column View (Ctrl+Shift+1)">
```
Comment (yaichenbaum) [inline Files/UserControls/NavigationToolbar.xaml:1176]:
```suggestion
                                        ToolTipService.ToolTip="Column View (Ctrl+Shift+6)">
```

**Comment by cinqmilleans** [comment]
@yair100 This pr changes the order of the layouts from smallest to largest (column / detail / tile / grid small-> medium-> large), in the context menu and in the navigationtoolbar (which does not currently display the same order) . This is the order that is used for the shortcuts. So I did not apply your changes.

**Comment by yair100** [comment]
> @yair100 This pr changes the order of the layouts from smallest to largest (column / detail / tile / grid small-> medium-> large), in the context menu and in the navigationtoolbar (which does not currently display the same order) . This is the order that is used for the shortcuts. So I did not apply your changes.

As discussed in the issue, we would prefer to keep the current order since it's arranged by what's used more often.

**Comment by cinqmilleans** [comment]
@yair100 Ok for me. The changes you proposed only modify the tooltips. I'll edit to keep the current order.

**Comment by yair100** [comment]
> @yair100 Ok for me. The changes you proposed only modify the tooltips. I'll edit to keep the current order.

Correct, that was just the general idea 🙂

**Comment by cinqmilleans** [comment]
@yair100 It is done.

**Review by yair100 — DISMISSED** [review]

**Comment by winston-de** [comment]
Looks good, but can you please resolve the merge conflicts?

**Comment by cinqmilleans** [comment]
@winston-de You are right. The merge is resolved.

**Review by winston-de — APPROVED** [review]
LGTM, nice work!

**Pull request merged**